jlm/llvm/opt/alias-analyses/Andersen.hpp

```diff
@@ -191,21 +176,24 @@ class Andersen final : public AliasAnalysis
     [[nodiscard]] static Configuration
     NaiveSolverConfiguration() noexcept
     {
-      auto config = Configuration();
-      config.EnableOfflineVariableSubstitution(false);
-      config.EnableOfflineConstraintNormalization(false);
-      config.SetSolver(Solver::Naive);
-      config.EnableOnlineCycleDetection(false);
-      return config;
+      return Configuration{};
```

Comment:
Is this a good practice?
A change of the default constructor now has an effect on what the NaiveSolverConfiguration() means.

Reply:
The solution I went with is not the cleanest, no. I have a comment about default values of members being the "naive" configuration, but it is probably cleaner to not have a default constructor at all, and force assigning every field.

The reason I didn't do this is that I ended up with constructor invocations that looked like `Configuration(false, false, Naive, LeastRecentlyFired, false, false, false, false, true)`.

Constructors are nicer, through, as we get errors if the signature changes.
Maybe I could add names to the values somehow, to make everything clearer?

Reply:
For the NaiveSolverConfiguration(), it could explicitly set the parameters the way the deleted version did, instead of relying on the default constructor default settings.

Reply:
I agree with Magnus here. Regarding the names for the constructor, C# had support for this:

c.foobar(
bar: false,
blub: 4,
....);

which was really nice. The closest to this you can get in C++ is comments:

c.foobar(
false, // bar
4,       // blub
...);